src/Plutarch/Orphans.hs

```diff
@@ -62,3 +102,165 @@ instance PTryFrom PData (PAsData PScriptHash) where
+                . encodeUtf8
+
+instance (Serialise a) => Aeson.ToJSON (AsBase16Codec a) where
+    toJSON =
+        Aeson.String
+            . encodeByteString
+            . toStrict
+            . serialise @a
+            . unAsBase16Codec
+
+instance (Serialise a) => Aeson.FromJSON (AsBase16Codec a) where
+    parseJSON v =
+        Aeson.parseJSON @Text v
+            >>= either
+                (parserThrowError [] . show)
+                (pure . AsBase16Codec)
+                . deserialiseOrFail
+                . fromStrict
+                . encodeUtf8
+
+deriving via (AsBase16Bytes TxId) instance Aeson.ToJSON TxId
+deriving via (AsBase16Bytes TxId) instance Aeson.FromJSON TxId
+
+deriving anyclass instance Aeson.ToJSON TxOutRef
+deriving anyclass instance Aeson.FromJSON TxOutRef
+
+deriving via
+    (AsBase16Bytes CurrencySymbol)
+    instance
+        (Aeson.ToJSON CurrencySymbol)
+deriving via
+    (AsBase16Bytes CurrencySymbol)
+    instance
+        (Aeson.FromJSON CurrencySymbol)
+
+deriving via
+    (AsBase16Bytes TokenName)
+    instance
+        (Aeson.ToJSON TokenName)
+deriving via
+    (AsBase16Bytes TokenName)
+    instance
+        (Aeson.FromJSON TokenName)
+
+deriving via
+    (AsBase16Bytes ValidatorHash)
+    instance
+        (Aeson.ToJSON ValidatorHash)
+deriving via
+    (AsBase16Bytes ValidatorHash)
+    instance
+        (Aeson.FromJSON ValidatorHash)
+
+deriving via
+    (AsBase16Bytes StakeValidatorHash)
+    instance
+        (Aeson.ToJSON StakeValidatorHash)
+deriving via
+    (AsBase16Bytes StakeValidatorHash)
+    instance
+        (Aeson.FromJSON StakeValidatorHash)
+
+deriving via
+    (AsBase16Codec StakeValidator)
+    instance
+        (Aeson.ToJSON StakeValidator)
+deriving via
+    (AsBase16Codec StakeValidator)
+    instance
+        (Aeson.FromJSON StakeValidator)
+
+deriving via
+    (AsBase16Bytes ScriptHash)
+    instance
+        (Aeson.ToJSON ScriptHash)
+deriving via
+    (AsBase16Bytes ScriptHash)
+    instance
+        (Aeson.FromJSON ScriptHash)
+
+deriving via
+    Integer
+    instance
+        (Aeson.ToJSON POSIXTime)
+deriving via
+    Integer
+    instance
+        (Aeson.FromJSON POSIXTime)
+
+deriving via
+    (AsBase16Bytes BuiltinByteString)
+    instance
+        (Aeson.ToJSON BuiltinByteString)
+deriving via
+    (AsBase16Bytes BuiltinByteString)
+    instance
+        (Aeson.FromJSON BuiltinByteString)
+
+deriving via
+    (AsBase16Codec Validator)
+    instance
+        (Aeson.ToJSON Validator)
+deriving via
+    (AsBase16Codec Validator)
+    instance
+        (Aeson.FromJSON Validator)
+
+deriving via
+    (AsBase16Codec MintingPolicy)
+    instance
+        (Aeson.ToJSON MintingPolicy)
+deriving via
+    (AsBase16Codec MintingPolicy)
+    instance
+        (Aeson.FromJSON MintingPolicy)
+
+deriving via
+    (AsBase16Codec Script)
+    instance
+        (Aeson.ToJSON Script)
+deriving via
+    (AsBase16Codec Script)
+    instance
+        (Aeson.FromJSON Script)
```

Comment:
All require `since`s.

Reply:
ooft. I noticed this, went to add to the change log, and then immediately forgot :sweat_smile: